modules/040-node-manager/hooks/update_node_group_status.go

```diff
@@ -351,6 +351,9 @@ func handleUpdateNGStatus(input *go_hook.HookInput) error {
 		}
 		statusMsg := fmt.Sprintf("%s %s", nodeGroup.Error, failureReason)
 		statusMsg = strings.TrimSpace(statusMsg)
+		// truncate to maximum allowed message limit
+		// https://github.com/kubernetes/kubernetes/blob/3e442b74f717ab2f43897d7af50de6114486e459/pkg/apis/core/validation/events.go#L38
+		statusMsg = statusMsg[:1024]
```

Comment:
if statusMsg is empty (and it could be) you will have a panic here. You should move it 1 line down, under `if len(statusMsg) > 0`

Reply:
Still won't be enough if it's shorter, but longer than zero. Fixed it!